### ISSUE TITLE
Default output for associative arrays

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#11455]
If you have an associative array `A` and you simply call `writeln(A)`, the output is not useful since it lists the values in arbitrary order and without the keys.

I'd propose that the default output should in some way indicate the key for each value.

**Comment by ben-albrecht** [comment]
I agree with the idea, but wonder if associative array `writeThis()` methods should stay consistent with other array types, and this behavior should be provided in the map-variant of associative arrays instead (https://github.com/chapel-lang/chapel/issues/9452).

**Comment by mppf** [comment]
> but wonder if associative array writeThis() methods should stay consistent with other array types

I'm not so sure about this because the default output is pretty un-interpretable. So what's the point?

> this behavior should be provided in the map-variant of associative arrays

that seems clear

**Comment by ben-albrecht** [comment]
> I'm not so sure about this because the default output is pretty un-interpretable. So what's the point?

Consistency with other array types. I don't think it's a particularly compelling reason to keep the current behavior, but just wanted to make sure it was mentioned.

**Comment by mppf** [comment]
I'd argue that the fact that the rectangular array types have useful `writeln` output and the associative types do not is a bigger problem with consistency

**Comment by mppf** [comment]
Some options:
 * add some kind of indication of the key for each value
 * sort the values but don't show what keys they correspond to (at least it is plausible that the output could be used this way)
 * make it an error (runtime? compilation?) to print out an associative array in this manner
 * leave it the way it is and just accept that printing an associative array on its own isn't useful; you have to write a zippered loop to get useful output
